### ISSUE TITLE
Although the model is generally successful, it sometimes labels completely irrelevant locations.

### ISSUE THREAD

**Issue by oztrkoguz** [ultralytics/ultralytics#18934]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am training object detection on 2200 images of 1920x1080 using the Yolov8s model. During the training process, the image size was set to 1920 and the batch size was set to 16, and training was performed for a total of 600 epochs. However, although the model's success is generally high, it occasionally labels completely irrelevant places. What could be the possible reasons for this problem and how can I solve it?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @oztrkoguz, thank you for bringing this up and for your detailed description of the issue 🚀! We recommend reviewing the [Docs](https://docs.ultralytics.com/) for guidance, particularly the [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) section, which might help you optimize your training further.

If this is a 🐛 Bug Report, please assist us in debugging it by providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/). This helps us better understand the circumstances causing the incorrect labeling behavior.

If this is a custom training ❓ Question, providing additional information would be greatly helpful, such as:
- Dataset details (e.g., class types, variability, and labeling practices used)
- Specific examples or screenshots of the erroneous predictions
- Training logs or validation metrics/results over the 600 epochs

Here are some general suggestions to troubleshoot and improve model performance:
1. Ensure data quality: Verify that your annotations are consistent and correct. Biases or errors in labeling can lead to the model learning incorrect patterns.
2. Evaluate the dataset balance: Imbalanced datasets (e.g., too few examples of certain classes) can lead to suboptimal performance.
3. Experiment with hyperparameters: Adjust the `lr0`, `imgsz`, or other training hyperparameters to see if they influence the results. A lower `confidence` threshold for predictions might also help reduce incorrect detections.
4. Multi-scale training: Consider enabling image size augmentations during training for better generalization.

## Upgrade

Make sure your environment is up to date! Upgrade to the latest `ultralytics` package with:

```bash
pip install -U ultralytics
```

This ensures you are working with the latest enhancements and bug fixes.

## Environments

If you'd like to test your model and environments on different platforms, consider running YOLO in any of these free and verified environments:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

Feel free to also join the Ultralytics community for advice and insights:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time discussions
- [Discourse](https://community.ultralytics.com/) for engaging in-depth conversations
- [Subreddit](https://reddit.com/r/Ultralytics) to exchange knowledge with fellow users

An Ultralytics engineer will assist you further soon. Thank you for your patience and for using Ultralytics YOLO ❤️!

**Comment by glenn-jocher** [comment]
@oztrkoguz thank you for sharing the details of your training setup! Incorrect labeling of irrelevant areas can often indicate issues with dataset quality or class imbalance. Please ensure all images are labeled accurately and consistently, and that the dataset includes sufficient variety to represent real-world conditions. Additionally, check for overfitting due to the high number of epochs by reviewing your validation metrics. If the issue persists, consider adding background images (images without objects) to help reduce false positives. For further insights, refer to our guide on [Tips for Best Training Results](https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/).

**Comment by Haseeeb21** [comment]
@oztrkoguz what is the probability of that irrelevant detections? 
Else you can also increase the confidence score of the model when predicting.

**Comment by oztrkoguz** [comment]
I will work on data diversity, thank you @glenn-jocher 
Thank you, I already tried it during the testing phase. But it didn't solve my problem because the free space detection percentage was high. @Haseeeb21

**Comment by glenn-jocher** [comment]
@oztrkoguz to address high false positives, consider adjusting both the confidence threshold and NMS IoU threshold during inference (`conf=0.5`, `iou=0.7` in your predict command). Verify label accuracy using the `train_batch*.jpg` mosaics generated during training, and ensure your dataset contains 5-10% background images as recommended in our [Training Tips Guide](https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/). If overfitting is suspected (600 epochs), review your validation curve metrics.

**Comment by Haseeeb21** [comment]
@glenn-jocher what is the difference btw confidence and iou? which is used / preferred where and which holds more importance?
and for background images, it can be any image with any background (floor, sky, etc.) and not the object we are training on, or should be like the background in the dataset no label file (txt) is required or it should be present but be empty?

**Comment by glenn-jocher** [comment]
@Haseeeb21 both confidence and IoU thresholds are important but serve different purposes. Confidence (`conf`) filters low-probability detections, while IoU (`iou`) merges overlapping boxes. For most cases, start with default values (`conf=0.25`, `iou=0.7`) and adjust based on your validation metrics ([see docs](https://docs.ultralytics.com/guides/model-evaluation-insights/)).

Background images should **not contain any objects** and require **no labels** (empty files or no files). They can include any non-object backgrounds like floors/sky. Add 5-10% background images as shown in our [Training Tips](https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/#background-images).
